src/cascadia/TerminalSettingsEditor/ViewModelHelpers.idl.h

```diff
@@ -10,4 +10,13 @@
         set;                                     \
     };                                           \
     Boolean Has##Name { get; };                  \
-    void Clear##Name()
+    void Clear##Name();                          \
+    Microsoft.Terminal.Settings.Model.Profile Name##OverrideSource { get; }
+
+#define PERMANENT_OBSERVABLE_PROJECTED_SETTING(Type, Name) \
```

Comment:
If you're going to introduce another macro that is a subset of the first one, cna you abstract them into a base macro like in the viewmodelhelpers C++ header?